contracts/modules/capital/Pool.sol

```diff
@@ -221,76 +221,34 @@ contract Pool is IPool, MasterAware, ReentrancyGuard {
   /* claim related functions */
 
   /**
-   * @dev Execute the payout in case a claim is accepted
-   * @param assetId        Index of the asset
+   * @dev Executes a payout
+   * @param assetIndex     Index of the payout asset
    * @param payoutAddress  Send funds to this address
    * @param amount         Amount to send
    */
-  function sendClaimPayout (
-    uint assetId,
+  function sendPayout (
+    uint assetIndex,
     address payable payoutAddress,
     uint amount
-  ) external onlyInternal nonReentrant returns (bool success) {
+  ) external override onlyInternal nonReentrant {
     bool ok;
-    Asset memory asset = assets[assetId];
+    Asset memory asset = assets[assetIndex];
 
-    // [todo] consider adding a minimum sum for triggering an MCR update to save gas
-    uint amountInETH;
     if (asset.assetAddress == ETH) {
       // solhint-disable-next-line avoid-low-level-calls
-      (ok, /* data */) = payoutAddress.call.value(amount)("");
-      amountInETH = amount;
+      (bool transferSucceeded, /* data */) = payoutAddress.call{value: amount}("");
+      require(transferSucceeded, "Pool: ETH transfer failed");
     } else {
-      ok =  _safeTokenTransfer(asset.assetAddress, payoutAddress, amount);
+      IERC20(asset.assetAddress).safeTransfer(payoutAddress, amount);
       uint rate = priceFeedOracle.getAssetToEthRate(asset.assetAddress);
       require(rate > 0, "Pool: Zero rate");
-      amountInETH = amount.mul(rate).div(1e18);
-    }
-
-    if (ok) {
-      emit Payout(payoutAddress, asset.assetAddress, amount);
-      uint totalAssetValue = getPoolValueInEth();
-      mcr.updateMCRInternal(totalAssetValue, true);
-    }
-
-    return ok;
-  }
-
-  /**
-   * @dev safeTransfer implementation that does not revert
-   * @param tokenAddress ERC20 address
-   * @param to destination
-   * @param value amount to send
-   * @return success true if the transfer was successfull
-   */
-  function _safeTokenTransfer (
-    address tokenAddress,
-    address to,
-    uint256 value
-  ) internal returns (bool) {
-
-    // token address is not a contract
-    if (!tokenAddress.isContract()) {
-      return false;
-    }
-
-    IERC20 token = IERC20(tokenAddress);
-    bytes memory data = abi.encodeWithSelector(token.transfer.selector, to, value);
-    // solhint-disable-next-line avoid-low-level-calls
-    (bool success, bytes memory returndata) = tokenAddress.call(data);
-
-    // low-level call failed/reverted
-    if (!success) {
-      return false;
     }
 
-    // tokens that don't have return data
-    if (returndata.length == 0) {
-      return true;
-    }
+    emit Payout(payoutAddress, asset.assetAddress, amount);
+    uint totalAssetValue = getPoolValueInEth();
 
-    // tokens that have return data will return a bool
-    return abi.decode(returndata, (bool));
+    // [todo] consider adding a minimum sum for triggering an MCR update to save gas
```

Comment:
Payouts happen less often, and given the complexity to calculate the payout amount in every asset it's probably simpler (and maybe even cheaper) just to update it.